### PR TITLE
Pet Buffs are save now!

### PR DESCRIPTION
**Pull request by dagochen**
Hopefully this works now.

### CODE REVIEW

**Comment by Phatcat** [comment]
Looks perfec'

Now, how do I reproduce this issue so that I can test your fix properly? =)

**Comment by dagochen** [comment]
1. Get a Hunter.
2. Get a pet.
3. Learn Mark of Wild or Fortitude or....
4. Buff your pet.
5. Dismiss your pet with your spell in your spellbook or by taking a flight.
6. Call your pet again.
7. The buffs on your pet are gone.

File: src/game/SpellEffects.cpp [hunk 1]
```diff
@@ -4285,7 +4285,7 @@ void Spell::EffectDismissPet(SpellEffectIndex /*eff_idx*/)
     if (!pet || !pet->isAlive())
         return;
 
-    pet->Unsummon(PET_SAVE_NOT_IN_SLOT, m_caster);
+    pet->Unsummon(PET_SAVE_AS_CURRENT, m_caster);
```
Comment (Phatcat) [inline src/game/SpellEffects.cpp]: You sure you don't wanna do a check for the type here as well and only do save as current on hunter pets and continue to do not in slot for the others. You tested to make sure it didn't break anything with dismissing warlock pets?

**Comment by dagochen** [comment]
The only spell with SPELL_EFFECT_DISMISS is Dismiss, which is a hunter only available ability (according to QSW).

**Comment by Phatcat** [comment]
Then your change is correct.

**Comment by dagochen** [comment]
This fixes only change behaviour for pet dismiss on spell, on movefollow being too far away and taking a flight. In case of putting pet into stable its like before and wont save the auras.

**Comment by Phatcat** [comment]
Probably we don't wanna save auras in the stable either so that seems okay to me.

It all sounds good to be honest, I just pulled your work locally and going to test it now. Building atm.

**Comment by Phatcat** [comment]
The second commit makes https://github.com/cmangos/mangos-classic/commit/50b9aa1290e663955e59816fa1215ade5aa484ca#diff-b4db11cbf2813cd24b3c165885fc08a7R4790
work.
Thank you.

**Comment by Phatcat** [comment]
How do you feel about squashing up the commits into 1? also would you maybe consider rewording the commit to have a short title; then you can fill up the commit message as much as you want =)

**Comment by dagochen** [comment]
Oh no squashing *afraid*....I can give it a try :D

**Comment by dagochen** [comment]
holy shit :D i made it! *cheer* thanks to your help last time phat ;) would not made it without you!

**Comment by Phatcat** [comment]
I shouldn't have been so quick to tell you to squash because I have a simplification for your code, give me a sec to type it up.

**Comment by dagochen** [comment]
np ;) I am in IRC so we dont need to spam here around.

**Comment by Phatcat** [comment]
Okay, I'm just going to apply the simplification myself when merging, no biggie, just looking to get this one moving along =)

**Comment by Phatcat** [comment]
Merged.

**Pull request closed without merging**